src/shared/openssl-util.h

```diff
@@ -5,6 +5,16 @@
 #include "macro.h"
 #include "sha256.h"
 
+typedef enum OpensslKeySourceType {
+        OPENSSL_KEY_SOURCE_FILE,
+        OPENSSL_KEY_SOURCE_ENGINE,
+        OPENSSL_KEY_SOURCE_PROVIDER,
+        _OPENSSL_KEY_SOURCE_MAX,
+        _OPENSSL_KEY_SOURCE_INVALID = -EINVAL,
+} OpensslKeySourceType;
```

Comment:
frnakly, even if this is in openssl-util.h i would drop the "Openssl_" namespacing prefix here. it doesn#t really make things prettier or clearer. And strictly speaking "file" isn't even an openssl concept.